packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -1122,7 +1150,16 @@ export const ContextualMenuBase: React.FunctionComponent<IContextualMenuProps> =
       hasCheckmarks: boolean,
       hasIcons: boolean,
     ): React.ReactNode => {
-      const { contextualMenuItemAs: ChildrenRenderer = ContextualMenuItem } = props;
+      let ChildrenRenderer: IComponentAs<IContextualMenuItemProps> = ContextualMenuItem;
+
+      if (item.contextualMenuItemAs) {
+        ChildrenRenderer = composeComponentAs(item.contextualMenuItemAs, ChildrenRenderer);
+      }
+
+      if (props.contextualMenuItemAs) {
+        ChildrenRenderer = composeComponentAs(props.contextualMenuItemAs, ChildrenRenderer);
+      }
```

Comment:
Should `props.contextualMenuItemAs` override `item.contextualMenuItemAs`? I think I'd lean towards the `item` prop taking precedence since it's more specific, but let me know if there's another reason to keep this order

Reply:
The way this was done in `CommandBar` was that the whole-bar override supersedes the `ContextualMenu` one. But I agree it would make more sense to flip it, or at least for there also to be the ability for there to be an override-able "base" component. This is worth a follow-up for that capability.